common/distribution/dist.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package distribution
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// DistType represents a distribution type
+type DistType string
+
+const (
+	DistScheme = "cimd"
+
+	// Distribution types
+	DistTypeAppc       DistType = "appc"
```

Comment:
Rather than specifying strings here, why not go with concrete distribution instances, and let the `Distribution` interface implement only behavior, i.e `Compare(Distribution) bool`?

Compare https://github.com/opencontainers/image-spec/blob/master/schema/schema.go#L21.